config.go

```diff
@@ -2159,10 +2170,14 @@ func configToFlatMap(cfg Config) (map[string]string, error) {
 			longName := fieldType.Tag.Get("long")
 			namespace := fieldType.Tag.Get("namespace")
 			group := fieldType.Tag.Get("group")
+			hidden := fieldType.Tag.Get("hidden")
+
+			var key string
+
 			switch {
 			// We have a long name defined, this is a config value.
 			case longName != "":
-				key := longName
+				key = longName
```

Comment:
Since this `case` is the only place where we actually have a value (and not a group where recurse into further), we should move the deprecation field check to here. That requires fewer lines of diff and should give us the same result.

Reply:
nice, done!